### PR TITLE
ipcache: fix expensive conflict logging for frequently used prefixes

### PR DESCRIPTION
**Pull request by squeed**
Searching for conflicts when a given prefixes has a lot of resources is very expensive. Specifically, all resources must be sorted before checking for conflicts.
    
Given that most resources do not set metadata that can conflict, this is not necessary. So, skip searching for conflicts if the supplied resource cannot cause a conflict.

```release-note
Fixes slow policy import times when many network policies reference the same CIDR.
```

### CODE REVIEW

File: pkg/ipcache/metadata_test.go [hunk 1]
```diff
@@ -1258,3 +1258,15 @@ func Test_metadata_mergeParentLabels(t *testing.T) {
...
 }
+
+func BenchmarkManyResources(b *testing.B) {
+	m := newMetadata()
+
+	prefix := netip.MustParsePrefix("1.1.1.1/32")
+	lbls := labels.GetCIDRLabels(prefix)
+
+	for i := range b.N {
+		resource := types.NewResourceID(types.ResourceKindCNP, fmt.Sprintf("namespace_%d", i), "my-policy")
```
Comment (nathanjsweet) [inline pkg/ipcache/metadata_test.go:1270]: Import `fmt`.

**Review by nathanjsweet — APPROVED** [review]

File: pkg/ipcache/metadata_test.go (hunk 1 above)
Comment (squeed) [reply pkg/ipcache/metadata_test.go:1270]: you can't tell me what to do

**Review by doniacld — APPROVED** [review]
LGTM!

**Pull request merged**